lib/data/repository/user_repository.dart

```diff
@@ -1,27 +1,30 @@
 import 'package:cobe_hive_mobile_app/data/network/cobe_api.dart';
 import 'package:cobe_hive_mobile_app/data/models/user.dart';
-import 'package:cobe_hive_mobile_app/providers/network_providers/cobe_api_provider.dart';
+import 'package:cobe_hive_mobile_app/providers/network_providers/connectivity_status_provider.dart';
 import 'package:dio/dio.dart';
+import 'package:flutter/foundation.dart';
+import 'package:hive_flutter/hive_flutter.dart';
 import 'package:hooks_riverpod/hooks_riverpod.dart';
 
 final userRepositoryProvider = Provider<UserRepository>(
-    (ref) => UserRepository(ref.read(cobeApiProvider)));
+    (ref) => UserRepository(ref.read(cobeApiProvider), ref));
 
 class UserRepository {
   final CobeApi _cobeApi;
-  UserRepository(this._cobeApi);
+  final Ref ref;
+
+  UserRepository(this._cobeApi, this.ref);
+
+  final _userBox = Hive.box<User>('users');
 
   Future<List<User>> getUsers() async {
-    try {
-      final response = await _cobeApi.getUsers();
-      final userList =
-          (response.data as List).map((e) => User.fromJson(e)).toList();
-      await Future.delayed(const Duration(seconds: 2));
-      return userList;
-    } on DioException catch (e) {
-      final errorMessage = e.toString();
-      throw errorMessage;
+    if (await ref
+            .read(connectivityStatusProvider.notifier)
+            .checkConnectivity() ==
+        ConnectivityStatus.isConnected) {
```

Comment:
improvment: separate this into a variable, it will look better 

await ref
            .read(connectivityStatusProvider.notifier)
            .checkConnectivity()